### ISSUE TITLE
Shows core dumped for memory.

### ISSUE THREAD

**Issue by mythio** [PawanKartikS/ASE#26]
I compiled a program
`MOV AL, 12`

and the results for memory broke. It shows
```
Memory:
free(): double free detected in tcache 2
[1]    23962 abort (core dumped)  ./ase test.asm --all-flags
```

Tried compiling it with and without flags and the issue persists.

**Comment by PawanKartikS** [comment]
That's weird. It works fine for me. 

Just confirming, are you sure you ran it with `./ase test.asm --all-flags`?

**Comment by PawanKartikS** [comment]
Can you post a screenshot?

**Comment by kalpaj12** [comment]
Try with `./ase test.asm --all-flags`

**Comment by PawanKartikS** [comment]
> Try with `./ase test.asm --all-flags`

That's what he tried it with. See the last line of the dump.

**Comment by kalpaj12** [comment]
there is an s missing in flags

**Comment by PawanKartikS** [comment]
But still, that shouldn't trigger a core dump.

**Comment by PawanKartikS** [comment]
@kalpaj12 Can you reproduce this issue? I cannot seem to. Works fine on my end.

**Comment by mythio** [comment]
![Screenshot from 2019-05-20 23-00-38](https://user-images.githubusercontent.com/32430197/58040431-2254de00-7b53-11e9-973d-f1ff77ca01fe.png)

**Comment by PawanKartikS** [comment]
@kalpaj12 Can you reproduce this issue?

**Comment by PawanKartikS** [comment]
@mythio Can you re-run the program but this time without any flags and share the result?

**Comment by PawanKartikS** [comment]
@mythio Follow this:

1) Open main.c
2) Comment out `destroy_table()` and `destroy_glob()`.
3) Recompile the emulator using the `build.sh` script, re-run the emulator and share the result here.

**Comment by kalpaj12** [comment]
nope, can't reproduce this

**Comment by mythio** [comment]
@PawanKartikS after commenting line 157, and 158, there is no core dump. But the memory outputs are not shown.

**Comment by PawanKartikS** [comment]
I'm guessing this has something to do with `destroy_glob()` because `destroy_table()` belongs to `tengine` and if there were something wrong with it, the asserts in `bind.c` should've failed. My best guess is that even though that the stack is not being used, it's memory is being `free`'d.

**Comment by kalpaj12** [comment]
okay, I agr

> I'm guessing this has something to do with `destroy_glob()` because `destroy_table()` belongs to `tengine` and if there were something wrong with it, the asserts in `bind.c` should've failed. My best guess is that even though that the stack is not being used, it's memory is being `free`'d.

I am going to make the same changes as in #25, when freeing the memory, @mythio  check and let me know if the issue persists. Clone my [fork](https://github.com/kalpaj12/ASE)

**Comment by PawanKartikS** [comment]
@mythio Clone @kalpaj12 's latest commit and see if the issue is fixed.

**Comment by mythio** [comment]
core dumped again

**Comment by kalpaj12** [comment]
@mythio , clone my fork and test again. I guess the latest commit must have resolved the issue

**Comment by PawanKartikS** [comment]
At the same time, please add a new line after the closing brace on line 151.

**Comment by kalpaj12** [comment]
for which file?

**Comment by PawanKartikS** [comment]
Main.c

**Comment by PawanKartikS** [comment]
Also, when you do that, truncate the comment on line 139.

**Comment by kalpaj12** [comment]
done, with this.

> At the same time, please add a new line after the closing brace on line 151.

and this.
> Also, when you do that, truncate the comment on line 139.

**Comment by kalpaj12** [comment]
Fixed in #27
